log4j-core/src/main/java/org/apache/logging/log4j/core/impl/Log4jLogEvent.java

```diff
@@ -786,7 +786,7 @@ public LogEventProxy(final LogEvent event, final boolean includeLocation) {
             this.thrownProxy = event.getThrownProxy();
             this.contextData = memento(event.getContextData());
             this.contextStack = event.getContextStack();
-            this.source = includeLocation ? event.getSource() : null;
+            this.source = includeLocation || !event.isIncludeLocation() ? event.getSource() : null;
```

Comment:
This line assumes that `event.getSource()` will not compute any location information if `isIncludeLocation()` is `false`. This is true for our implementations of `LogEvent`, but is not required by the specification (Javadoc).

Reply:
`LogEventProxy` is not a relevant class in 3.x anymore due to https://issues.apache.org/jira/browse/LOG4J2-3228